pocs/filterwheel/filterwheel.py

```diff
@@ -0,0 +1,224 @@
+        """ Connect to filter wheel """
+        raise NotImplementError
+
+    def move_to(self, position, blocking=False):
+        """
+        Move the filter wheel to the given position.
+
+        The position can be expressed either as an integer, or as (part of) one of the names from
+        the filter_names list. To allow filter names of the form '<filter band>_<serial number>'
+        to be selected by band only position can be a substring from the start of one
+        of the names in the filter_names list, provided that this produces only one match.
+
+        Args:
+            position (int or str): position to move to.
+            blocking (bool, optional): If False (default) return immediately, if True block until
+                the filter wheel move has been completed.
+
+        Returns:
+            threading.Event: Event that will be set to signal when the move has completed
+        """
+        assert self.is_connected, self.logger.error("Filter wheel must be connected to move")
+        position = self._parse_position(position)
+        self.logger.info("Moving {} to position {} ({})".format(
+            self, position, self.filter_names[position - 1]))
+        move_event = threading.Event()
+
+        if position == self.position:
+            # Don't go nowhere
+            move_event.set()
+            return move_event
+
+        self._move_to(position, move_event)  # Private method to actually perform the move.
+
+        if blocking:
+            move_event.wait()
+
+        return move_event
+
+##################################################################################################
+# Private methods
+##################################################################################################
+
+    def _move_to(self, position, move_event):
+        raise NotImplementedError
+
+    def _parse_position(self, position):
+        """
+        Converts a requested position to an integer filter wheel position.
+
+        If position is a string it will search the list of filter names for one that begins with
+        that string and return the corresponding integer position, otherwise (or if there is no
+        match) it will do an explicity case to an integer.
+        """
+        int_position = None
+        if isinstance(position, str):
+            # Got a string, so search for a match in the filter names list
+            for i, filter_name in enumerate(self.filter_names):
+                if filter_name.startswith(position):
+                    if int_position is None:
+                        int_position = i + 1  # 1 based numbering for filter wheel positions
+                    else:
+                        # Already matched at least once
+                        msg = "More than one filter name matches '{}', using '{}'".format(
+                            position, self.filter_names[int_position - 1])
+                        self.logger.warning(msg)
+                        break
```

Comment:
Purely so that I can log a warning about multiple matches. Ambigious filter wheel `move_to()` commands probably indicate a mistake somewhere, so shouldn't pass entirely silently.